marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx

```diff
@@ -37,7 +38,225 @@ describe('RTL Source-to-entity map tests', () => {
+
+       //Reset the search filter on Entity table
+       fireEvent.click(entityfilterIcon);
+       let resetEntitySearch = getByTestId('ResetSearch-name');
+       expect(resetEntitySearch).toBeInTheDocument();
+       fireEvent.click(resetEntitySearch);
+ 
+       //Check if the expected values are present now after resetting the filter on Entity table.
+       expect(getByText('propId')).toBeInTheDocument();
+       expect(getByText('propName')).toBeInTheDocument();
+    });
+
+    test('Column option selector in Entity table',() => {
+      
+      const { getByText,getByTestId,queryByText } = render(<SourceToEntityMap {...data.mapProps} 
+        mappingVisible={true} 
+        entityTypeProperties={data.mapProps.entityTypePropertiesUpdated} 
+        sourceData={data.mapProps.sourceDataUpdated}/>);
+
+      //Set the data for texting in xpath expression
+
+      let exp = getByText('testNameInExp');
+      expect(exp).toBeInTheDocument();
+      fireEvent.change(exp, { target: {value: "concat(propName,'-NEW')" }});
+      fireEvent.blur(exp);
+      expect(getByText("concat(propName,'-NEW')")).toBeInTheDocument();
+      
+      /* Test column option selector in Entity table  */
+      let colOptSelect = getByText('Column Options');
+      expect(colOptSelect).toBeInTheDocument();
+      fireEvent.click(colOptSelect);
+      let Name = getByTestId('columnOptionsCheckBox-name');
+      let Type = getByTestId('columnOptionsCheckBox-type');
+      let XPathExpression = getByTestId('columnOptionsCheckBox-key');
+      let Value = getByTestId('columnOptionsCheckBox-value');
+      expect(Name).toBeInTheDocument();
+      expect(Type).toBeInTheDocument();
+      expect(XPathExpression).toBeInTheDocument();
+      expect(Value).toBeInTheDocument();
+
+      fireEvent.click(Name); //Uncheck Name column
+
+      //Confirm that the Name column is not visible in Entity table anymore.
+      expect(queryByText('propId')).not.toBeInTheDocument();
+      expect(queryByText('propName')).not.toBeInTheDocument();
+
+      fireEvent.click(XPathExpression); //Uncheck Name column
+
+      //Confirm that the XPath Expression column(Along with Name Column) is not visible in Entity table anymore.
+      expect(queryByText("concat(propName,'-NEW')")).not.toBeInTheDocument();
+      expect(queryByText('propId')).not.toBeInTheDocument();
+      expect(queryByText('propName')).not.toBeInTheDocument();
+
+      //Checking the columns one by one in selector and verify that they appear in entity table
+      fireEvent.click(Name); //Check Name column
+      //Props below should be available now
+      expect(getByText('propId')).toBeInTheDocument(); 
+      expect(getByText('propName')).toBeInTheDocument();
+
+      fireEvent.click(XPathExpression); //Check XPathExpression column
+      //Props below should be available now
+      expect(getByText("concat(propName,'-NEW')")).toBeInTheDocument();
+
+    });
+
+    test('Sorting in Source and Entity table',() => {
```

Comment:
sorting is good. 
like the spacing & grouping of columns, makes it readable. Goes well with "Arrange, Act, Assert" way of writing tests.